python/cudf/cudf/core/column/column.py

```diff
@@ -2211,13 +2251,10 @@ def _copy_type_metadata_from_arrow(
     * When `arrow_array` is decimal type and `cudf_column` is
     Decimal64Dtype, copy precisions.
     """
-    if pa.types.is_decimal(arrow_array.type) and isinstance(
-        cudf_column, cudf.core.column.DecimalColumn
-    ):
-        cudf_column.dtype.precision = arrow_array.type.precision
-    elif pa.types.is_struct(arrow_array.type) and isinstance(
-        cudf_column, cudf.core.column.StructColumn
-    ):
+    cudf_dtype = _cudf_dtype_from_arrow_type(arrow_array.type)
+    cudf_column = cudf_column._apply_type_metadata(cudf_dtype)
```

Comment:
I'm still confused as to why we need further processing of `cudf_column` after this step?

Reply:
Think we would need to include some additional logic in `_copy_type_metadata` for the different recursive calls done for copying arrow struct or list arrays, but this should be doable.